### DISCUSSION TITLE
Check JPath queries for '' instead of "

### DISCUSSION THREAD

**Discussion by TysvdHeuvel** [StefanMaron/BusinessCentral.LinterCop#764] [New Rule]
We tend to use JPath a lot. So, a line of code like this is very common: 
```
MagentoCustomer.SelectToken('$.custom_attributes[?(@.attribute_code == ''activation_status'')].value', MagentoActivationStatusToken);
```
But JPath expressions can become quite complex, so therefore you need to test them. And this is where the problem starts. Most of these easy to use online JPath builders/validators all use " (double quotes) to wrap strings. But BC somehow can't handle that. The expression then simply has no result. So you always have to remember not to copy/paste the expression, but also to replace all double quotes with 2 single quotes. 

Having a rule that flags this, would have saved me a ton of hours.

**Comment by Arthurvdv** [comment]
@TysvdHeuvel , thank you for the idea!

I've created a new rule: [LC0080 - Replace double quotes in JPath expressions with two single quotes.](https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0080) which is now available in the [v0.31.4](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.31.4) (pre)release version of the LinterCop.

The rule is currently straight forward; If there's a double quote in the Path, throw an Warning. For now I've bound the rule exclusivity on `SelectToken` method on `JsonToken` objects. 

If there are exceptions need on this, please let me know.

P.S. Do you by any chance now of this is also relevant on `XPath` methods on `XmlNode`/`XmlDocument` objects?

**Reply by TysvdHeuvel** [reply]
Hi @Arthurvdv,

Cool! I'll test it ASAP. 
Don't know about Xpath. But my guess is that that works the same way.

Thanks!

**Comment by jwikman** [comment]
I've never seen that issue with XML and XPath, so I don't think the rule should include those.

**Comment by Arthurvdv** [comment]
The rule is now available in the release version [v0.32.0](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.32.0) of the LinterCop.
